db/dcrsqlite/sqlite.go

```diff
@@ -982,27 +984,34 @@ func (db *DB) RetrievePoolValAndSizeRange(ind0, ind1 int64) ([]float64, []float6
 
 // RetrieveAllPoolValAndSize returns all the pool values and sizes stored since
 // the first value was recorded up current height.
-func (db *DB) RetrieveAllPoolValAndSize() (*dbtypes.ChartsData, error) {
-	chartsData := new(dbtypes.ChartsData)
+func (db *DB) RetrieveAllPoolValAndSize(chartsData *dbtypes.ChartsData) (*dbtypes.ChartsData, error) {
```

Comment:
This is confusing since this chartsData is a merger of two different charts data sets.  The caller has to know this and prepare the inputs, and tease apart the output.  I think this would be better taking the three slices as inputs, and returning the modified (or new) slices as outputs.